spyder/plugins/framesexplorer/widgets/main_widget.py

```diff
@@ -228,7 +229,9 @@ def create_new_widget(self, shellwidget):
         widget.sig_hide_finder_requested.connect(self.hide_finder)
         widget.sig_update_actions_requested.connect(self.update_actions)
 
-        widget.sig_show_namespace.connect(shellwidget.set_namespace_view)
+        widget.sig_show_namespace.connect(
+            lambda namespace: self.sig_show_namespace.emit(
+                namespace, shellwidget))
```

Comment:
When I merged your initial PR for Frames I didn't understand why you added an action to show the namespace of the current frame.

I mean, my question: why can't we do that automatically?

Reply:
Let's say I capture the state of the program. I have 3 threads with a few frames. If I want to see the namespace of the last frame of thread number two, I can show it in the variable explorer